# -*- mode: org; mode: auto-fill; fill-column: 80 -*-

#+TITLE: httplib.HTTPSConnection Insecure
#+OPTIONS:   d:t
#+LINK_UP:  ./
#+LINK_HOME: ../

* Issue

  As the Python documentation makes clear, [[http://docs.python.org/library/httplib.html#httplib.HTTPSConnection][httplib.HTTPSConnection]] is insecure.
  It doesn't validate or verify the server's certificate, meaning that we can't
  guarntee the server's identity or even that we're consistently connecing to
  the same server.

* Searches

  This issue is easy to find.  There may be false positives, because it's
  possible to verify certificates around the connection itself, but the fact
  that it's there means there's something to investigate.

  #+begin_src sh
    grep -nHr "HTTPSConnection" ../src/*
  #+end_src

  #+results:

** Code

   | Code                                     |
   |------------------------------------------|
   | [[../src/connectors/cli/controller.py::174]] |
   | [[../src/connectors/cli/controller.py::181]] |
   | [[../src/connectors/cli/controller.py::202]] |

   The only place this exists in code is in the cli controller.  This is
   corrected by [[file:16.org][adding reasonable interprocess communication]].

** Tests

   | Tests                                          |
   |------------------------------------------------|
   | [[../src/connectors/https/test_controller.py::54]] |
   | [[../src/connectors/cli/test_controller.py::128]]  |

   I don't know if I should be alright with leaving this in any tests, but
   there it is, of course, less urgent.


* Fixes [0/1]

  Remove or Secure Occurrences.

** TODO Secure occurrences.

   [[file:16.org][Add reasonable interprocess communication]].

* Discussion

* Metadata
  :PROPERTIES:
  :Status:   Incomplete
  :Priority: 100
  :Owner:    Nick Daly
  :Tags:     Security
  :Blocked:  [[file:16.org][Interprocess Communication]]
  :Description:     HTTPSConnection Insecure
  :END:
